### ISSUE TITLE
mix archive.install hex phx_new 1.4.0 fails with existing, old install and requires mix archive.uninstall phx_new first

### ISSUE THREAD

**Issue by rikkus** [elixir-lang/elixir#8541]
This can be avoided by running ```mix archive.uninstall phx_new``` first, but the warnings and error don't make it obvious that this is what will fix it.

### Environment

```
Erlang/OTP 20 [erts-9.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.6.4 (compiled with OTP 20)

MacOS Mojave 10.14.1

Darwin WMAC3EFG8WL 18.2.0 Darwin Kernel Version 18.2.0: Fri Oct  5 19:41:49 PDT 2018; root:xnu-4903.221.2~2/RELEASE_X86_64 x86_64
```

### Current behavior
```
> mix archive.install hex phx_new 1.4.0
Resolving Hex dependencies...
Dependency resolution completed:
New:
  phx_new 1.4.0
* Getting phx_new (Hex package)
Compiling 10 files (.ex)

warning: redefining module Mix.Tasks.Phx.New.Ecto (current version loaded from /Users/rik/.mix/archives/phx_new/phx_new/ebin/Elixir.Mix.Tasks.Phx.New.Ecto.beam)
  lib/mix/tasks/phx.new.ecto.ex:1

[... lots of similar warnings elided ...]

== Compilation error in file lib/phx_new/ecto.ex ==
** (File.Error) could not read file "/Users/chris/oss/phoenix/installer/templates/phx_ecto/repo.ex": no such file or directory
    (elixir) lib/file.ex:310: File.read!/1
    lib/phx_new/generator.ex:28: anonymous fn/4 in Phx.New.Generator."MACRO-__before_compile__"/2
    (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
    lib/phx_new/generator.ex:24: anonymous fn/3 in Phx.New.Generator."MACRO-__before_compile__"/2
    (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
    expanding macro: Phx.New.Generator.__before_compile__/1
    lib/phx_new/ecto.ex:1: Phx.New.Ecto (module)

```

### Expected behavior

```
> mix archive.install hex phx_new 1.4.0
Resolving Hex dependencies...
Dependency resolution completed:
New:
  phx_new 1.4.0
* Getting phx_new (Hex package)
Compiling 10 files (.ex)
Generated phx_new app
Generated archive "phx_new-1.4.0.ez" with MIX_ENV=prod
Are you sure you want to install "phx_new-1.4.0.ez"? [Yn]
* creating /Users/rik/.mix/archives/phx_new-1.4.0
```

**Comment by ericmj** [comment]
I think the issue is that we have the old archive version still in the code path. So we need to make sure to remove it from the code path and unload any modules from it.

**Comment by josevalim** [comment]
This has been fixed in v1.7.4. :)

**Comment by ericmj** [comment]
Wow, I knew I recalled this issue https://github.com/elixir-lang/elixir/commit/c88137330e2da0527f60b9a37cc5732d7b12e798.
